crates/oxc_linter/src/rules/typescript/no_magic_numbers.rs

```diff
@@ -0,0 +1,896 @@
+			  [K in keyof Other]: `${K & number}`;
+			};
+			      ",
+            Some(serde_json::json!([{ "ignoreTypeIndexes": true, "ignore": [0, 3] }])),
+        ),
+    ];
+
+    let fail = vec![
+        ("var foo = 42", enforce_const.clone()), // { "ecmaVersion": 6 },
+        ("var foo = 0 + 1;", None),
+        // ("var foo = 42n", enforce_const.clone()), // {                "ecmaVersion": 2020            },
+        // ("var foo = 0n + 1n;", None), // {                "ecmaVersion": 2020            },
+        ("a = a + 5;", None),
+        ("a += 5;", None),
+        ("var foo = 0 + 1 + -2 + 2;", None),
+        (
+            "var foo = 0 + 1 + 2;",
+            Some(serde_json::json!([{                "ignore": [0, 1]            }])),
+        ),
+        (
+            "var foo = { bar:10 }",
+            Some(serde_json::json!([{                "detectObjects": true            }])),
+        ),
+        ("console.log(0x1A + 0x02); console.log(071);", None), // {                "sourceType": "script"            },
+        (
+            "var stats = {avg: 42};",
+            Some(serde_json::json!([{                "detectObjects": true            }])),
+        ),
+        ("var colors = {}; colors.RED = 2; colors.YELLOW = 3; colors.BLUE = 4 + 5;", None),
+        ("function getSecondsInMinute() {return 60;}", None),
+        ("function getNegativeSecondsInMinute() {return -60;}", None),
+        ("var data = ['foo', 'bar', 'baz']; var third = data[3];", None),
+        ("var data = ['foo', 'bar', 'baz']; var third = data[3];", Some(serde_json::json!([{}]))),
+        (
+            "var data = ['foo', 'bar', 'baz']; var third = data[3];",
+            Some(serde_json::json!([{"ignoreArrayIndexes": false}])),
+        ),
+        ("foo[-100]", ignore_array_indexes.clone()),
+        ("foo[-1.5]", ignore_array_indexes.clone()),
+        ("foo[-1]", ignore_array_indexes.clone()),
+        ("foo[-0.1]", ignore_array_indexes.clone()),
+        ("foo[-0b110]", ignore_array_indexes.clone()), // { "ecmaVersion": 2015 },
+        ("foo[-0o71]", ignore_array_indexes.clone()),  // { "ecmaVersion": 2015 },
+        ("foo[-0x12]", ignore_array_indexes.clone()),
+        ("foo[-012]", ignore_array_indexes.clone()), // { "sourceType": "script" },
+        ("foo[0.1]", ignore_array_indexes.clone()),
+        ("foo[0.12e1]", ignore_array_indexes.clone()),
+        ("foo[1.5]", ignore_array_indexes.clone()),
+        ("foo[1.678e2]", ignore_array_indexes.clone()),
+        ("foo[56e-1]", ignore_array_indexes.clone()),
+        ("foo[5.000000000000001]", ignore_array_indexes.clone()),
+        ("foo[100.9]", ignore_array_indexes.clone()),
+        ("foo[4294967295]", ignore_array_indexes.clone()),
+        ("foo[1e300]", ignore_array_indexes.clone()),
+        ("foo[1e310]", ignore_array_indexes.clone()),
+        ("foo[-1e310]", ignore_array_indexes.clone()),
+        // ("foo[-1n]", ignore_array_indexes.clone()), // { "ecmaVersion": 2020 },
```

Comment:
Forgot to add checks for big int literals?

Reply:
I did not find a good way to match from `serde_json` the f64 and string. When I try to cast f64 to a string with `.to_string()` or `format!()` the NumberBase will be converted to 10. 

When BigInt Package was removed some other rules was outcommented.  
If this is needed, I need some help to parse the `serde_json` values right :)

Reply:
Okay after another try and using enum I could solve almost every problem. The `is_array_index` function is a big problem because BigInts has a NumberBase too. 
I need an Parser for it for checking:

- is it negative
- Is it a real BigInt Number or a BigInt Float
- does it extend the max array index spec: `number < f64::from(u32::MAX)`

@DonIsaac Please tell me if the tests are a MUST for this PR :)

Reply:
@camc314 opinion?

Reply:
so basically we know that it's a BigInt, but can't tell wiether it is negative/real/float/< u32 max?

i think its fine for now. i can't see people indexing into arrays with big ints very often